### PR TITLE
100% doc coverage in package:flutter_driver

### LINKED ISSUE #5883: flutter_driver docs
About 90 members in flutter_driver are lacking docs.

cc @yjbanov

### PR DESCRIPTION
**Pull request by yjbanov**
I added docs in most places. A few members were converted to private.

Fixes #5883

/cc @Hixie

### CODE REVIEW

File: packages/flutter_driver/lib/src/error.dart [hunk 1]
```diff
@@ -47,16 +52,39 @@ void _log(LogLevel level, String loggerName, Object message) {
...
 class LogRecord {
   const LogRecord._(this.level, this.loggerName, this.message);
 
+  /// The severity of the log record.
   final LogLevel level;
+
+  /// The name of the logger that logged the message.
   final String loggerName;
+
+  /// The log message.
```
Comment (Hixie) [inline packages/flutter_driver/lib/src/error.dart]: Are there conventions for how to format this string? e.g. capitalisation, punctuation

File: packages/flutter_driver/lib/src/error.dart [hunk 2]
```diff
@@ -47,16 +52,39 @@ void _log(LogLevel level, String loggerName, Object message) {
...
   final String message;
 
+  /// Short description of the log level.
```
Comment (Hixie) [inline packages/flutter_driver/lib/src/error.dart]: Human-readable, or machine-readable, or debugging aid only?

File: packages/flutter_driver/lib/src/error.dart [hunk 3]
```diff
@@ -65,26 +93,33 @@ class LogRecord {
 
 /// Package-private; users should use other public logging libraries.
 class Logger {
+  /// Creates a new logger.
   Logger(this.name);
 
+  /// Logger name.
```
Comment (Hixie) [inline packages/flutter_driver/lib/src/error.dart]: Needs more detail. Any time you feel tempted to write docs that could have been written by someone who knew nothing other than the fully-qualified name of the identifier, you need more detail. :-)

File: packages/flutter_driver/lib/src/extension.dart [hunk 4]
```diff
@@ -88,11 +88,13 @@ class FlutterDriverExtension {
...
   }
 
-  final WidgetController prober = new WidgetController(WidgetsBinding.instance);
+  final WidgetController _prober = new WidgetController(WidgetsBinding.instance);
   final Map<String, CommandHandlerCallback> _commandHandlers = <String, CommandHandlerCallback>{};
   final Map<String, CommandDeserializerCallback> _commandDeserializers = <String, CommandDeserializerCallback>{};
   final Map<String, FinderConstructor> _finders = <String, FinderConstructor>{};
 
+  /// Processes a driver command configured by [params] and returns a result
+  /// as a JSON object.
```
Comment (Hixie) [inline packages/flutter_driver/lib/src/extension.dart]: This should document the shape of the returned JSON object

File: packages/flutter_driver/lib/src/find.dart [hunk 5]
```diff
@@ -91,6 +105,11 @@ abstract class SerializableFinder {
     throw new DriverError('Unsupported search specification type $finderType');
   }
 
+  /// Serializes common fields to JSON.
+  ///
+  /// It is expected that overrides call `super.serialize` and add more fields
```
Comment (Hixie) [inline packages/flutter_driver/lib/src/find.dart]: grammar is dubious in this sentence

File: packages/flutter_driver/lib/src/gesture.dart [hunk 6]
```diff
@@ -5,12 +5,15 @@
...
 
+/// Taps on a target widget located by [finder].
 class Tap extends CommandWithTarget {
   @override
   final String kind = 'tap';
 
+  /// Creates a tap command to tap on a widget located by [finder].
   Tap(SerializableFinder finder) : super(finder);
 
+  /// Deserializes this command from JSON.
```
Comment (Hixie) [inline packages/flutter_driver/lib/src/gesture.dart]: Anywhere where you currently say "Deserializes this command from JSON.", you should either describe the expected shape of the JSON input, or point to the API that generates the expected JSON in the first place.

File: packages/flutter_driver/lib/src/timeline.dart [hunk 7]
```diff
@@ -9,6 +9,7 @@
 ///
 /// See https://docs.google.com/document/d/1CvAClvFfyA5R-PhYUmn5OOQtYMH4h6I0nSsKchNAySU/preview
 class Timeline {
+  /// Creates a timeline given JSON-encoded timeline data.
```
Comment (Hixie) [inline packages/flutter_driver/lib/src/timeline.dart]: here and below, you should talk about the JSON shape or point to the encoder

File: packages/flutter_driver/lib/src/timeline_summary.dart [hunk 8]
```diff
@@ -21,6 +21,7 @@ const Duration kBuildBudget = const Duration(milliseconds: 8);
 
 /// Extracts statistics from the event loop timeline.
```
Comment (Hixie) [inline packages/flutter_driver/lib/src/timeline_summary.dart]: this should probably point to `[Timeline]`.

**Comment by Hixie** [comment]
This is great!

**Comment by Hixie** [comment]
LGTM

File: packages/flutter_driver/lib/src/error.dart [hunk 9]
```diff
@@ -65,26 +93,33 @@ class LogRecord {
 
 /// Package-private; users should use other public logging libraries.
```
Comment (yjbanov) [inline packages/flutter_driver/lib/src/error.dart]: Done.

File: packages/flutter_driver/lib/src/error.dart (hunk 3 above)
Comment (yjbanov) [inline packages/flutter_driver/lib/src/error.dart]: Done.

File: packages/flutter_driver/lib/src/extension.dart (hunk 4 above)
Comment (yjbanov) [inline packages/flutter_driver/lib/src/extension.dart]: Done.

File: packages/flutter_driver/lib/src/find.dart (hunk 5 above)
Comment (yjbanov) [inline packages/flutter_driver/lib/src/find.dart]: Done.

File: packages/flutter_driver/lib/src/gesture.dart (hunk 6 above)
Comment (yjbanov) [inline packages/flutter_driver/lib/src/gesture.dart]: Done (pointed to the corresponding `serialize` methods).

File: packages/flutter_driver/lib/src/input.dart [hunk 10]
```diff
@@ -35,22 +44,29 @@ class SetInputTextResult extends Result {
   Map<String, dynamic> toJson() => <String, dynamic>{};
 }
 
+/// Submits text entered in a text input widget.
```
Comment (yjbanov) [inline packages/flutter_driver/lib/src/input.dart]: Done (pointed to https://docs.flutter.io/flutter/material/Input/onSubmitted.html)

File: packages/flutter_driver/lib/src/timeline.dart (hunk 7 above)
Comment (yjbanov) [inline packages/flutter_driver/lib/src/timeline.dart]: The link to the format is on the class` dartdoc, but I moved it here.

File: packages/flutter_driver/lib/src/timeline_summary.dart (hunk 8 above)
Comment (yjbanov) [inline packages/flutter_driver/lib/src/timeline_summary.dart]: Done.

**Pull request merged**